xcdat/spatial_avg.py

```diff
@@ -338,78 +339,136 @@ def _get_weights(
-                    pmb_index = None
-
-                # Perform differencing to get weights and add metadata.
-                dim_wts = np.abs(dom_bounds[:, 1] - dom_bounds[:, 0])
-
-                # If pmb_index is not None it means that an extra
-                # longitude bound was added and the weight vector
-                # is too long. This extra weight should be added to the
-                # grid cell that crossed the prime meridian (and then
-                # removed from the dim_wts vector).
-                if pmb_index is not None:
-                    dim_wts[pmb_index] = dim_wts[pmb_index] + dim_wts[-1]
-                    dim_wts = dim_wts[0:-1]
-
-                # ensure attributes remain on dim_wts
-                dim_wts.attrs = dom_bounds.attrs
-                dim_wts.name = dim + "_wts"
-                axis_weights[dim] = dim_wts
+        for axes, bounds_by_type in bounds.items():
+            d_bounds = bounds_by_type["domain"]
+            self._validate_domain_bounds(d_bounds)
+
+            r_bounds = bounds_by_type["region"]
+            if r_bounds is not None:
+                r_bounds = np.array(r_bounds, dtype="float")
+
+            if axes == "lon":
+                weights = self._get_longitude_weights(d_bounds, r_bounds)
+            elif axes == "lat":
+                weights = self._get_latitude_weights(d_bounds, r_bounds)
+
+            weights.attrs = d_bounds.attrs
+            weights.name = axes + "_wts"
+            axis_weights[axes] = weights
 
         weights = self._combine_weights(axis, axis_weights)
         return weights
 
-    def _align_longitude_to_360_axis(self, domain_bounds: xr.DataArray):
+    def _get_longitude_weights(
+        self, domain_bounds: xr.DataArray, region_bounds: np.array
+    ) -> xr.DataArray:
+        """Gets weights for the longitude axes.
+
+        This method performs longitudinal processing including (in order):
+
+        1. Aligning the axes orientations of the domain and region bounds to
+           (0, 360) to ensure compatibility in proceeding operations.
+        2. Handling grid cells that cross the prime meridian (e.g., [-1, 1])
+           by recreating the axes with two additional grid cells ([0, 1] and
+           [359, 360]) to ensure alignment with the (0, 360) axes orientation.
+           The prime meridian cell is returned as a variable for a proceeding
+           function.
+        3. Scaling the domain down to a region (if selected).
+        4. Calculating weights using the domain bounds.
+        5. If the prime meridian cell exists, use this cell's index to handle the
+           weights vector's increased length as a result of the two additional
+           grid cells. The extra weights are added to the prime meridian grid
+           cell and removed from the weights vector to ensure the lengths of the
+           weights and its corresponding domain remain in alignment.
```

Comment:
The inline comments have been moved to the docstrings of the methods with some edits. Feel free to suggest any changes here.